kevm_pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -16,23 +17,77 @@
     KRequire,
     KRewrite,
     KRule,
+    KSequence,
     KSort,
     KTerminal,
     KToken,
     KVariable,
 )
+from pyk.kastManip import buildRule, substitute
 from pyk.ktool import KPrint, paren
+from pyk.prelude import intToken
 from pyk.utils import intersperse
 
+from .utils import (
+    abstract_cell_vars,
+    build_empty_configuration_cell,
+    infGas,
+    kevmAccountCell,
+)
+
 
 def solc_compile(contract_file: Path) -> Dict[str, Any]:
     subprocess_res = subprocess.run([
         'solc', '--combined-json', 'abi,bin-runtime,storage-layout,hashes', str(contract_file),
-    ], check=True, capture_output=True)
+    ], capture_output=True)
+
+    if subprocess_res.returncode != 0:
+        fatal(f'solc error:\n{subprocess_res.stderr.decode()}')
 
     return json.loads(subprocess_res.stdout)
 
 
+def gen_spec_modules(kompiled_directory: Path):
+    kevm = KPrint(str(kompiled_directory))
+    kevm.symbolTable = kevmSymbolTable(kevm.symbolTable)
+    production_labels = [prod.klabel for module in kevm.definition.modules for prod in module.productions if prod.klabel is not None]
+    contract_names = [prod_label[9:] for prod_label in production_labels if prod_label.startswith('contract_')]
+    contract_function_labels = ['function_' + contract_name for contract_name in contract_names]
+    top_level_rules = [rule for module in kevm.definition.modules for rule in module.rules if type(rule.body) is KRewrite]
+    contract_function_signatures = [rule.body.lhs for rule in top_level_rules if type(rule.body.lhs) is KApply and rule.body.lhs.label in contract_function_labels]
+    spec_modules = []
+    for contract_name in contract_names:
```

Comment:
Consider extracting the body of this loop to a function.

Reply:
Done.